lib/jsonapi/plugs/deserializer.ex

```diff
@@ -0,0 +1,66 @@
+defmodule JSONAPI.Deserializer do
+  @moduledoc """
+  This plug "deserializes" params for ease of use when casting to changesets.
+  As a result, you are able to pattern match specific attributes in your controller
+  actions.
+
+  Note that this Plug will only deserialize your payload when the request's content
+  type is for a JSON:API request (i.e. "application/vnd.api+json"). All other
+  content types will be ignored.
+
+  ## Example
+
+  For example these params:
+      %{
+        "data" => %{
+          "id" => "1",
+          "type" => "user",
+          "attributes" => %{
+            "foo-bar" => true
+          },
+          "relationships" => %{
+            "baz" => %{"data" => %{"id" => "2", "type" => "baz"}}
+          }
+        }
+      }
+
+  are transformed to:
+
+      %{
+        "id" => "1",
+        "type" => "user"
+        "foo-bar" => true,
+        "baz-id" => "2"
```

Comment:
Great idea!  Added a test.  I'm not sure though there is a way to process other than this...

https://github.com/jeregrine/jsonapi/pull/222/commits/738c2b9e085120f194306261d4a70fc8ee161598#diff-ced812e7235b283a4598ca0ae957ecd0R91

Otherwise, down in this file, I have tests for `includes` with many types.